vnext/Microsoft.ReactNative/ReactApplication.h

```diff
@@ -19,19 +25,32 @@ namespace winrt::Microsoft::ReactNative::implementation {
 // It must be updated if the shape of generated ReactApplication_base is changed in future.
 // The only difference is that this class has no default constructor.
 template <typename D, typename... I>
-struct __declspec(empty_bases) NoDefaultCtorReactApplication_base
-    : implements<
-          D,
-          Microsoft::ReactNative::ReactApplication,
-          composable,
-          composing,
-          xaml::IApplicationOverrides,
-          xaml::IApplicationOverrides2,
-          I...>,
-      impl::require<D, xaml::IApplication, xaml::IApplication2, xaml::IApplication3>,
-      impl::base<D, xaml::Application>,
-      xaml::IApplicationOverridesT<D>,
-      xaml::IApplicationOverrides2T<D> {
+struct __declspec(empty_bases) NoDefaultCtorReactApplication_base : implements<
+                                                                        D,
+                                                                        Microsoft::ReactNative::ReactApplication,
+                                                                        composable,
+                                                                        composing,
+                                                                        xaml::IApplicationOverrides,
+#ifndef USE_WINUI3
+                                                                        xaml::IApplicationOverrides2,
```

Comment:
These sorts of differences are concerning... Do you have a sense of how widespread it is?

Reply:
WinUI 3 flattened the API so you'll see interfaces like `IUIElement7`, etc. that don't exist in WinUI 3 anymore, so it's pretty widespread.

Reply:
Maybe we should consolidate where possible (eg have a header file with the interface aliases).  It won't be possible in all cases (like this one), just trying to find ways to reduce/share instances of #ifdef USE_WINUI3.

---
In reply to: [445211208](https://github.com/microsoft/react-native-windows/pull/5309#discussion_r445211208) [](ancestors = 445211208)